subprojects/docs/src/docs/userguide/authoring-builds/build_lifecycle.adoc

```diff
@@ -15,37 +15,53 @@
 [[build_lifecycle]]
 = Build Lifecycle
 
-We said earlier that the core of Gradle is a language for dependency based programming. In Gradle terms this means that you can define tasks and dependencies between tasks. Gradle guarantees that these tasks are executed in the order of their dependencies, and that each task is executed only once. These tasks form a http://en.wikipedia.org/wiki/Directed_acyclic_graph[Directed Acyclic Graph]. There are build tools that build up such a dependency graph as they execute their tasks. Gradle builds the complete dependency graph _before_ any task is executed. This lies at the heart of Gradle and makes many things possible which would not be possible otherwise.
+Gradle is an example of *dependency based programming*: you define tasks and dependencies between tasks.
+Gradle guarantees that these tasks execute in the order of their dependencies.
+Your build scripts configure this dependency graph.
+This page discusses the phases of the lifecycle Gradle passes through as it interprets those scripts.
+Additionally, this page explains how you can react to events of the lifecycle using notifications.
 
-Your build scripts configure this dependency graph. Therefore they are strictly speaking _build configuration scripts_.
+== Task Graphs
+
+Some build tools assemble a task graph lazily, as they execute tasks.
+Gradle builds the complete task graph eagerly, before executing any task.
+
+Within each project, tasks form a http://en.wikipedia.org/wiki/Directed_acyclic_graph[Directed Acyclic Graph] (DAG).
+
+This diagram shows two example task graphs: one abstract and the other concrete.
+The diagram represents dependencies between tasks as arrows:
+
+.Two examples of Gradle task graphs
+image::task-dag-examples.png[Example task graphs]
+
+Both plugins and your own build scripts contribute to the task graph via the <<tutorial_using_tasks#sec:task_dependencies,task dependency mechanism>>.
 
 [[sec:build_phases]]
-== Build phases
+== Build Phases
 
 A Gradle build has three distinct phases.
+Gradle runs these phases in order: first initialization, then configuration, and finally execution.
 
 Initialization::
-Gradle supports single and multi-project builds. During the initialization phase, Gradle determines which projects are going to take part in the build, and creates a link:{groovyDslPath}/org.gradle.api.Project.html[Project] instance for each of these projects.
+- Detects the settings file.
+- Evaluates the <<organizing_gradle_projects#sec:settings_file,settings file>> to determine which projects participate in the build.
+- Creates a link:{groovyDslPath}/org.gradle.api.Project.html[`Project`] instance for every project.
 Configuration::
-During this phase the project objects are configured. The build scripts of _all_ projects which are part of the build are executed.
+- Evaluates the build scripts of every project participating in the build.
+- Creates a task graph for all projects.
```

Comment:
```suggestion
- Creates a task graph for requested tasks.
```

Reply:
Hm, maybe I'm misunderstanding here. Based on your previous comment, it sounds like this is true for projects that 100% use lazy configuration. Is this misleading for projects that don't use lazy configuration?

Reply:
There's a nuance here.
The task graph will always contain requested tasks and their dependencies only. Lazy task configuration or not.
In order to populate the task graph, Gradle will create and configure all the required task instances.

For all other tasks, not required in the task graph, if they are lazily configured then they won't be configured at all. If they are eagerly configured they are always configured when evaluating the Project.

Does it clarifies things?